### PR TITLE
Manually fix several for loops

### PR DESCRIPTION
**Pull request by nyurik**
Manually fix several for loops

I used this replacement, and then went through each and reasoned if wrapping can ever happen (mostly it cannot, e.g. -1 is ok if for loop starts at 1)

When reviewing, make sure to use "ignore blanks" because lots of code was only de-indented.

```diff
@@
expression idx, start, end;
@@
-   idx = start;
-   while idx < end
+   for idx in start..end
    {
        {
        ...
        }
-        idx = idx.wrapping_add(1);
    }
    
@@
expression idx, start, end;
@@
-   idx = start;
-   while idx <= end
+   for idx in start..=end
    {
        {
        ...
        }
-        idx = idx.wrapping_add(1);
    }
```

### CODE REVIEW

File: src/enc/brotli_bit_stream.rs [hunk 1]
```diff
@@ -846,21 +846,15 @@ fn BrotliStoreHuffmanTreeOfHuffmanTreeToBitMask(
...
-                    kHuffmanBitLengthHuffmanCodeBitLengths[l],
-                    kHuffmanBitLengthHuffmanCodeSymbols[l] as u64,
-                    storage_ix,
-                    storage,
-                );
-            }
-            i = i.wrapping_add(1);
-        }
+
+    for i in skip_some..codes_to_store {
```
Comment (danielrh) [inline src/enc/brotli_bit_stream.rs:850]: What if skip_some > codes_to_store...that would panic, right?
please fix or revert

**Review by danielrh — CHANGES_REQUESTED** [review]

File: src/enc/brotli_bit_stream.rs [hunk 2]
```diff
@@ -1705,17 +1682,12 @@ fn StoreTrivialContextMap(
...
         histogram[repeat_code] = num_types as u32;
-        histogram[0] = 1u32;
-        i = context_bits;
-        while i < alphabet_size {
-            {
-                histogram[i] = 1u32;
-            }
-            i = i.wrapping_add(1);
+        histogram[0] = 1;
+        for i in context_bits..alphabet_size {
```
Comment (danielrh) [inline src/enc/brotli_bit_stream.rs:1689]: what if context_bits >= alphabet_size...is that possible?

File: src/enc/brotli_bit_stream.rs [hunk 3]
```diff
@@ -1773,29 +1745,19 @@ fn MoveToFront(v: &mut [u8], index: usize) {
...
         return;
     }
     max_value = v_in[0];
-    i = 1;
-    while i < v_size {
-        {
-            if v_in[i] > max_value {
-                max_value = v_in[i];
-            }
+    for i in 1..v_size {
```
Comment (danielrh) [inline src/enc/brotli_bit_stream.rs:1754]: are we sure v_size > 0?

File: src/enc/entropy_encode.rs [hunk 4]
```diff
@@ -75,52 +75,40 @@ pub fn SortHuffmanTreeItems<Comparator: HuffmanComparator>(
...
-                    if {
-                        let _old = j;
-                        j = j.wrapping_sub(1);
-                        _old
-                    } == 0
-                    {
-                        break;
-                    }
+    if n < 13 {
+        for i in 1..n {
```
Comment (danielrh) [inline src/enc/entropy_encode.rs:79]: what if n == 0...is that a possibility?

**Review by danielrh — CHANGES_REQUESTED** [review]

File: src/enc/entropy_encode.rs [hunk 5]
```diff
@@ -75,52 +75,40 @@ pub fn SortHuffmanTreeItems<Comparator: HuffmanComparator>(
...
                 let gap: usize = gaps[g as usize];
-                let mut i: usize;
-                i = gap;
-                while i < n {
-                    {
-                        let mut j: usize = i;
-                        let mut tmp: HuffmanTree = items[i];
-                        while j >= gap
-                            && (comparator.Cmp(&mut tmp, &mut items[j.wrapping_sub(gap)]))
+                for i in gap..n {
```
Comment (danielrh) [inline src/enc/entropy_encode.rs:102]: could gap be > n?

File: src/enc/entropy_encode.rs [hunk 6]
```diff
@@ -262,18 +249,11 @@ pub fn BrotliOptimizeHuffmanCountsForRle(
...
-                while i < length.wrapping_sub(1) {
-                    {
-                        if counts[i.wrapping_sub(1)] != 0u32
-                            && (counts[i] == 0u32)
-                            && (counts[i.wrapping_add(1)] != 0u32)
-                        {
-                            counts[i] = 1u32;
-                        }
+            if zeros < 6 {
+                for i in 1..length.wrapping_sub(1) {
```
Comment (danielrh) [inline src/enc/entropy_encode.rs:253]: is length always 2 or more here?

**Comment by nyurik** [comment]
in a for loop, if `start > end`, the loop will be skipped, just like with the `while` loop code before. It seems all your comments are about that, so might not be applicable?

**Comment by danielrh** [comment]
> in a for loop, if `start > end`, the loop will be skipped, just like with the `while` loop code before. It seems all your comments are about that, so might not be applicable?

Ahh yes you are correct... in that case approve

**Review by danielrh — APPROVED** [review]
Rebase looks good

**Pull request merged**